triggers/service/src/main/scala/com/digitalasset/daml/lf/engine/trigger/TriggerRunner.scala

```diff
@@ -51,19 +51,23 @@ object TriggerRunner {
     config.withTriggerLogContext { implicit triggerContext =>
       Behaviors.setup { ctx =>
         // Spawn a trigger runner impl. Supervise it. Stop immediately on
-        // initialization halted exceptions, retry any initialization or
-        // execution failure exceptions.
+        // initialization halted and trigger overflow exceptions, retry
+        // any other initialization or execution failure exceptions.
         val runner =
           ctx.spawn(
             Behaviors
               .supervise(
                 Behaviors
                   .supervise(
                     Behaviors
-                      .supervise(TriggerRunnerImpl(config))
-                      .onFailure[InitializationHalted](stop)
+                      .supervise(
+                        Behaviors
+                          .supervise(TriggerRunnerImpl(config))
+                          .onFailure[InitializationHalted](stop)
+                      )
+                      .onFailure[UnauthenticatedException](stop)
                   )
-                  .onFailure[UnauthenticatedException](stop)
+                  .onFailure[TriggerOverflowException](stop)
```

Comment:
Feel its safer to have an explicit stop for overflow exceptions rather than relying on current observations from testing